src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/CompiledPageRouteModelProvider.cs

```diff
@@ -17,14 +18,24 @@ public class CompiledPageRouteModelProvider : IPageRouteModelProvider
         private readonly object _cacheLock = new object();
         private readonly ApplicationPartManager _applicationManager;
         private readonly RazorPagesOptions _pagesOptions;
+        private readonly ILogger<CompiledPageRouteModelProvider> _logger;
         private List<PageRouteModel> _cachedModels;
 
         public CompiledPageRouteModelProvider(
             ApplicationPartManager applicationManager,
             IOptions<RazorPagesOptions> pagesOptionsAccessor)
+            : this(applicationManager, pagesOptionsAccessor, loggerFactory: null)
+        {
+        }
+
+        public CompiledPageRouteModelProvider(
+            ApplicationPartManager applicationManager,
+            IOptions<RazorPagesOptions> pagesOptionsAccessor,
+            ILoggerFactory loggerFactory)
         {
             _applicationManager = applicationManager;
             _pagesOptions = pagesOptionsAccessor.Value;
```

Comment:
Not directly related to your changes, but "pagesOptionsAccessor" can be null, and that would throw here.

Reply:
We've historically not had null checks in ctors that are meant to be DI-ed and types not callable via user code.